### PR TITLE
fix: Don't use debug rendering by default in BodyComponent

### PR DESCRIPTION
**Pull request by spydon**
# Description

Since debug rendering now renders on top of everything (since ages), `BodyComponent` shouldn't hijack the debug render method for its normal rendering.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame_forge2d/lib/body_component.dart [hunk 1]
```diff
@@ -14,18 +14,17 @@ import 'sprite_body_component.dart';
 /// it is a good idea to turn on [debugMode] for it so that the bodies can be
 /// seen
 abstract class BodyComponent<T extends Forge2DGame> extends Component
-    with HasGameRef<T> {
+    with HasGameRef<T>, HasPaint {
```
Comment (erickzanardo) [inline packages/flame_forge2d/lib/body_component.dart:17]: Excellent! I was going to suggest that but I forgot to raise the issue, glad you did it here!

**Review by erickzanardo — COMMENTED** [review]
Overall LGTM 🚀, but there is a different issue now, children of body component are rendered upside down.

To reproduce this, try adding a `TextComponent` inside a body component.

**Comment by st-pasha** [comment]
> but there is a different issue now, children of body component are rendered upside down.

Is this problem caused by this PR, or was it there before?

**Comment by spydon** [comment]
> > but there is a different issue now, children of body component are rendered upside down.
> 
> Is this problem caused by this PR, or was it there before?

If it was added directly as a child to the `BodyComponent` I think the problem was already there before, I think it should be fixed now though (this whole mess really needs a refactoring though).

**Review by wolfenrain — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**